### PR TITLE
[8.x](backport #40678) [Metricbeat][vSphere] Support for configurable IntervalId for performance API

### PR DESCRIPTION
**Pull request by kush-elastic**
Re-Raised from: https://github.com/elastic/beats/pull/40744
- Enhancement

The vSphere Performance API offers support for Intervals and Levels, allowing users to collect various types of counters and metrics from vSphere environments. Depending on the configured Interval, the API provides detailed or aggregated performance data from vSphere.

For more information, refer to the following resources:
- [Performance Data Collection](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-web-services-programming-guide/GUID-481756FA-9F0B-4768-8E3B-EAFCEEC908B2.html)
- [Data Collection Intervals](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.monitoring.doc/GUID-247646EA-A04B-411A-8DD4-62A3DCFCF49B.html)
- [Data Collection Levels](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.monitoring.doc/GUID-25800DE4-68E5-41CC-82D9-8811E27924BC.html)

We can add support for configuration that help users collect the specific performance data they need.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/obs-integration-team/issues/36

<hr>This is manually backport of pull request #40678.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -327,10 +326,21 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Add new metricset network for the vSphere module. {pull}40559[40559]
 - Add new metricset resourcepool for the vSphere module. {pull}40456[40456]
 - Log the total time taken for GCP `ListTimeSeries` and `AggregatedList` requests {pull}40661[40661]
+- Add `use_kubeadm` config option in kubernetes module in order to toggle kubeadm-config api requests {pull}40086[40086]
+- Log the total time taken for GCP `ListTimeSeries` and `AggregatedList` requests {pull}40661[40661]
+- Add new metrics for the vSphere Host metricset. {pull}40429[40429]
```
Comment (ishleenk17) [inline CHANGELOG.next.asciidoc:331]: Is this used from the last commit in main/8.15 ?

Comment (kush-elastic) [reply CHANGELOG.next.asciidoc:331]: This is after something merged to 8.x.

Comment (kush-elastic) [reply CHANGELOG.next.asciidoc:346]: I think i should remove them

Comment (ishleenk17) [reply CHANGELOG.next.asciidoc:342]: Yes, lets go as and when whatever was merged.

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**